znai-server/src/main/java/com/twosigma/znai/server/support/DocumentationSupportHandler.java

```diff
@@ -17,26 +17,27 @@
 package com.twosigma.znai.server.support;
 
 import com.twosigma.znai.utils.JsonUtils;
-import io.vertx.core.Handler;
 import io.vertx.core.http.HttpServerRequest;
 import io.vertx.core.http.HttpServerResponse;
 
-public class DocumentationSupportHandler implements Handler<HttpServerRequest> {
-    private final String docId;
-    private final String actor;
+public class DocumentationSupportHandler {
+    private static SupportMetaProvider supportMetaProvider;
 
-    public DocumentationSupportHandler(String docId, String actor) {
-        this.docId = docId;
-        this.actor = actor;
+    public static void setProvider(SupportMetaProvider supportMetaProvider) {
```

Comment:
I may have confused you with my previous comment. I am still fine with using `ServiceLoader` for extensions and de-coupling.

Reply:
I want to set it explicitly with a setter. If I have multiple implementation of SupportMetaProvider, ServiceLoader will load all of them.

Reply:
Fair point. The way TS extends znai internally is by wrapping a `cli` start method and more like a hard mode of extending it. 
Initial goal was to extend by only using `ServiceLoaders`. So that's why I want to stick to the `ServiceLoader` based plugin system. 
Do we have already other examples that extend by using setters?

Reply:
TsDocumentationCliAppExtension.initUploadHandlers() calls OnUploadFinishedServerHandlers.add(new DocUploadHander()) and OnRemoveFinishedServerHandlers.add(new DocRemoveHandler()).
As there is no hashCode() in the classes, the handlers might be executed twice.
I need to delete them.

Znai in Twosigma gets a support link from cmdb now, but it might change to get one from a db later. If I create a new SupportMetaProvider to get one from a db, the code will break it. I need to delete the old class or modify the class.

Reply:
It uses `add` instead of `ServiceLoaders` partially due to how `META-INF/services` Is not a first class citizen and partially that we already wrap `cli`. 
Not following about `hashCode`. 

If you change an implementation later, you would change `META-INF/services` content. Additionally you may have a guard preventing from having multiple implementations loaded. 

I feel like I am dragging this review. It is not my intention and I apologize it takes us this long. 
I need to start documenting internal architecture and intentions to reduce the level of assumptions.

Reply:
Now I understand why you mentioned `ServiceLoaders` a couple of times. I thought that `ServiceLoaderUtils` used 
 reflection to detect all of classes which implemented an interface.

I sent you an invitation from google hangout. If you find it, please accept it. Communicating over github is kind of slow for me, when I have a question.

Reply:
Do we still need this handler?

Reply:
No, I need SupportMeta and SupportMetaProvider, not Handler.
I deleted it.

Reply:
I thought we talked about using `UrlContentHandlers`. Could you please remind me how are we going to use `SupportMetaProvider` now?

Reply:
Customized `UrlContentHandler` will use `SupportMetaProvider` to return `SupportMeta`.

Reply:
Do you see a use case for `SupportMetaProvider` outside of your overrides?
Will I have a choice to either add link value to `meta.json` or override `SupportMetaProvider`? Or is it only being used on your side?
I feel like you can reduce it to just overriding `UrlContentHandler`, but I maybe missing something.

Reply:
Ok, I can put everything into `UrlContentHandler`.